web/src/views/InvestigationDetail.vue

```diff
@@ -498,17 +505,84 @@ export default defineComponent({
       }));
     }
 
+    async function toggleNonTiledImageOverlay(pin: Pin, dataset: Dataset) {
+      if (!dataset.content) return;
+      if (!featureLayer) return; // sanity check. this should exist
+      if (!nonTiledOverlayFeature) {
+        nonTiledOverlayFeature = featureLayer.createFeature('quad');
+      }
+      const quadData = nonTiledOverlayFeature.data() || [];
+      // eslint-disable-next-line
+      const existingOverlay = quadData.find((overlay: any) => overlay.pinId === pin.id);
+      if (existingOverlay) {
+        nonTiledOverlayFeature.data(
+          // eslint-disable-next-line
+          nonTiledOverlayFeature.data().filter((overlay: any) => overlay.pinId !== pin.id),
+        ).draw();
+        nonTiledOverlayFeature.draw();
+      } else {
+        try {
+          const urlRoot = process.env.VUE_APP_API_ROOT;
+          const url = `${urlRoot}/datasets/tile_source/${dataset.id}/thumbnail.png`;
+          const image: HTMLImageElement = await getNonTiledImage(url);
+          image.crossOrigin = 'use-credentials';
+          if (store.state.axiosInstance) {
+            const imageMetadata = await (await store.state.axiosInstance.get(`${urlRoot}/datasets/tile_source/${dataset.id}/metadata`)).data;
+            const ul = postGisToPoint(pin.child_location) || { x: 0, y: 0 };
+            const lr = {
+              x: ul.x + (imageMetadata.sizeX || 0),
+              y: ul.y + (imageMetadata.sizeY || 0),
+            };
+            quadData.push({
+              ul,
+              lr,
+              image,
+              pinId: pin.id,
+            });
+            nonTiledOverlayFeature.data(quadData).draw();
+          }
+        } catch (error) {
+          // eslint-disable-next-line
+          console.error(`Failed to show overlay for dataset ${dataset.id}.`);
+        }
+      }
+    }
+
+    function toggleDatasetPin(pin: Pin) {
```

Comment:
What's the purpose of this function? Why does it have a single-case switch statement (rather than just an if statement)?

Reply:
I was thinking ahead to supporting other `dataset_type`s. For example, pins that display tiled data would have a different `dataset_type`, and would be displayed differently than these small image pins. This function would take a pin and call the corresponding toggling function. The switch does look silly now with just one case, so I could remove this if you'd like. Just let me know.

Reply:
I think maybe under the new design we've discussed, this might improve generally (for the record, the problem is not the single case; it's the need for a switch statement at all).